### PR TITLE
Tests Boot 3 upgrade recipe descriptor

### PR DESCRIPTION
**Pull request by BoykoAlex**
Test cached recipe descriptor

### CODE REVIEW

**Comment by BoykoAlex** [comment]
Note that the test fails due to a bug in rewrite recipe class. Essentially it tests recipe descriptor creation in rewrite core. Once rewrite PR https://github.com/openrewrite/rewrite/pull/3042 is merged it should pass

File: src/test/java/org/openrewrite/java/spring/RecipeDescriptorTest.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.openrewrite.Recipe;
+import org.openrewrite.config.Environment;
+import org.openrewrite.config.RecipeDescriptor;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class RecipeDescriptorTest {
```
Comment (timtebeek) [inline src/test/java/org/openrewrite/java/spring/RecipeDescriptorTest.java:26]:
```suggestion
class RecipeDescriptorTest {
```

File: src/test/java/org/openrewrite/java/spring/RecipeDescriptorTest.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    private static Environment env;
+
+    @BeforeAll
+    public static void setupAll() {
+        env = Environment.builder().scanRuntimeClasspath().build();
+    }
+
+    @Test
+    public void createRecipeTest() throws Exception {
```
Comment (timtebeek) [inline src/test/java/org/openrewrite/java/spring/RecipeDescriptorTest.java:36]:
```suggestion
    void createRecipeTest() throws Exception {
```

File: src/test/java/org/openrewrite/java/spring/RecipeDescriptorTest.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import org.openrewrite.config.RecipeDescriptor;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class RecipeDescriptorTest {
+
+    private static Environment env;
+
+    @BeforeAll
+    public static void setupAll() {
```
Comment (timtebeek) [inline src/test/java/org/openrewrite/java/spring/RecipeDescriptorTest.java:31]:
```suggestion
    static void setupAll() {
```

File: src/test/java/org/openrewrite/java/spring/RecipeDescriptorTest.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        env = Environment.builder().scanRuntimeClasspath().build();
+    }
+
+    @Test
+    public void createRecipeTest() throws Exception {
+        Recipe r = env.listRecipes().stream().filter(d -> "org.openrewrite.java.spring.boot3.UpgradeSpringBoot_3_0".equals(d.getName())).findFirst().orElse(null);
```
Comment (timtebeek) [inline src/test/java/org/openrewrite/java/spring/RecipeDescriptorTest.java:37]:
```suggestion
        Recipe r = env.listRecipes().stream().filter(d -> "org.openrewrite.java.spring.boot3.UpgradeSpringBoot_3_0".equals(d.getName())).findFirst().orElseThrow();
```

File: src/test/java/org/openrewrite/java/spring/RecipeDescriptorTest.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        RecipeDescriptor recipeDescriptor = r.getDescriptor();
+        assertNotNull(recipeDescriptor);
+
+        assertEquals("org.openrewrite.java.spring.boot3.UpgradeSpringBoot_3_0", recipeDescriptor.getName());
+        assertEquals(
+          "Migrate applications to the latest Spring Boot 3.0 release. This recipe will modify an application's build files, make changes to deprecated/preferred APIs, and migrate configuration settings that have changes between versions. This recipe will also chain additional framework migrations (Spring Framework, Spring Data, etc) that are required as part of the migration to Spring Boot 2.7.\n"
+            + ""
+            + "",
```
Comment (timtebeek) [inline src/test/java/org/openrewrite/java/spring/RecipeDescriptorTest.java:45]:
```suggestion
          "Migrate applications to the latest Spring Boot 3.0 release. This recipe will modify an application's build files, make changes to deprecated/preferred APIs, and migrate configuration settings that have changes between versions. This recipe will also chain additional framework migrations (Spring Framework, Spring Data, etc) that are required as part of the migration to Spring Boot 2.7.\n",
```

**Review by timtebeek — APPROVED** [review]

**Comment by timtebeek** [comment]
Reran the tests and added some suggestions; would you like to see this merged @BoykoAlex or was it mostly to verify this once?

**Review by timtebeek — APPROVED** [review]

**Pull request merged**